### PR TITLE
Fix #16463 - 'REFERENCES' privilege checkbox's title

### LINKED ISSUE #16463: "REFERENCES" priviledge is used by MySQL - wrong message
### Steps to reproduce

1. Go to `Edit privileges` page of any user
2. Hover mouse over `REFERENCES` priviledge
4. See wrong message - at least MySQL 8 supports `REFERENCES` priviledge (without this permission, foreign keys can not be created, thus very misleading)

### Expected behavior

Message is shown only for databases that really does NOT support `REFERENCES` priviledge.

### Screenshots

image

### PR DESCRIPTION
**Pull request by prashanttholia**
### Description

Sets different title messages depending on database and database version for 'REFERENCES' checkbox in user 'Edit Preferences' form.

Fixes #16463

Before submitting pull request, please review the following checklist:

- [x] Make sure you have read our [CONTRIBUTING.md](https://github.com/phpmyadmin/phpmyadmin/blob/master/CONTRIBUTING.md) document.
- [ ] Make sure you are making a pull request against the correct branch. For example, for bug fixes in a released version use the corresponding QA branch and for new features use the _master_ branch. If you have a doubt, you can ask as a comment in the bug report or on the mailing list.
- [x] Every commit has proper `Signed-off-by` line as described in our [DCO](https://github.com/phpmyadmin/phpmyadmin/blob/master/DCO). This ensures that the work you're submitting is your own creation.
- [x] Every commit has a descriptive commit message.
- [x] Every commit is needed on its own, if you have just minor fixes to previous commits, you can squash them.
- [x] Any new functionality is covered by tests.

### CODE REVIEW

File: templates/server/privileges/privileges_table.twig [hunk 1]
```diff
@@ -642,7 +642,7 @@
         {%- trans 'Has no effect in this MySQL version.' %}"{{ row['References_priv'] == 'Y' ? ' checked' }}>
       <label for="checkbox_References_priv">
         <code>
-          <dfn title="{% trans 'Has no effect in this MySQL version.' %}">
+          <dfn title="{{ is_mariadb ? 'Not used for MariaDB servers.'|trans : mysql_int_version >= 50622 ? 'Required for creating foreign key relations.'|trans : 'Not used for this MySQL version.'|trans }}">
```
Comment (mvorisek) [inline templates/server/privileges/privileges_table.twig:645]:
```suggestion
          <dfn title="{{ is_mariadb ? 'Not used for MariaDB servers.'|trans : (mysql_int_version >= 50622 ? 'Required for creating foreign key relations.'|trans : 'Not used for this MySQL version.'|trans) }}">
```

**Review by williamdes — APPROVED** [review]
This looks very nice, I am making some changes to use the `Query\Compatiblity` class

**Comment by williamdes** [comment]
I will merge this soon :)
let me know if you have anything to say

File: templates/server/privileges/privileges_table.twig [hunk 2]
```diff
@@ -642,7 +642,8 @@
         {%- trans 'Has no effect in this MySQL version.' %}"{{ row['References_priv'] == 'Y' ? ' checked' }}>
       <label for="checkbox_References_priv">
         <code>
-          <dfn title="{% trans 'Has no effect in this MySQL version.' %}">
+          {# l10n: The "REFERENCES" privilege when granting privileges #}
+          <dfn title="{{ supports_references_privilege ? 'Required for creating foreign key relations.'|trans : (is_mariadb ? 'Not used on MariaDB.'|trans: 'Not used for this MySQL version.'|trans) }}">
```
Comment (mvorisek) [inline templates/server/privileges/privileges_table.twig:646]:
```suggestion
          <dfn title="{{ supports_references_privilege ? 'Allows creating foreign key relations.'|trans : (is_mariadb ? 'Not used on MariaDB.'|trans: 'Not used for this MySQL version.'|trans) }}">
```

to be consistent with options above

Comment (williamdes) [reply templates/server/privileges/privileges_table.twig]: Thanks

**Comment by williamdes** [comment]
Merged into next 5.1 version
Thank you @prashanttholia !
(you can delete and re-fork if you intend to make another contribution. I had to re-write master branch to a QA one for this PR, next time please use a new branch for each PR ;) )

**Pull request merged**